examples/pytorch/image-classification/mnist.ipynb

```diff
@@ -133,43 +114,28 @@
     "        )\n",
     "    )\n",
     "\n",
-    "    # Create the model and load it into the device\n",
+    "    # Create the model and load it into the device.\n",
     "    device = torch.device(f\"{device}:{local_rank}\")\n",
     "    model = nn.parallel.DistributedDataParallel(Net().to(device))\n",
+    "    optimizer = torch.optim.SGD(model.parameters(), lr=0.1, momentum=0.9)\n",
     "\n",
     "    # Retrieve the Fashion-MNIST dataset\n",
-    "    if local_rank == 0:\n",
-    "        # Only download the dataset from local rank 0\n",
-    "        dataset = datasets.FashionMNIST(\n",
-    "            \"./data\",\n",
-    "            train=True,\n",
-    "            download=True,\n",
-    "            transform=transforms.Compose([transforms.ToTensor()]),\n",
-    "        )\n",
-    "        dist.barrier()\n",
-    "    else:\n",
-    "        # Wait for local rank 0 to complete downloading the dataset and load it\n",
-    "        dist.barrier()\n",
-    "        dataset = datasets.FashionMNIST(\n",
-    "            \"./data\",\n",
-    "            train=True,\n",
-    "            download=False,\n",
-    "            transform=transforms.Compose([transforms.ToTensor()]),\n",
-    "        )\n",
```

Comment:
I removed the dataset logic from this example for now. Since this Notebook demonstrates **single** proc per Node we should be good.

I think, we can find more elegant way to download data only on LOCAL_RANK=0 process.

WDYT @astefanutti ?

Reply:
Good idea, I agree we can revisit this and find a better approach.